src/components/20-molecules/datepicker/index.js

```diff
@@ -0,0 +1,369 @@
+    window.addEventListener('click', e => this.handleBodyClick(e));
+    window.setTimeout(() => {
+      window.addEventListener('resize', this.debounce(() => this.handleViewportCheck(this.inputField), 250));
+      window.addEventListener('scroll', this.debounce(() => this.handleViewportCheck(this.inputField), 250));
+      this.handleViewportCheck(this.inputField);
+    }, 100);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    window.axaComponents = window.axaComponents || {};
+    this.addEventListener('click', e => this.handleDatepickerClick(e));
+  }
+
+  disconnectedCallback() {
+    this.dropdownMonth.removeEventListener('AXA_CHANGE', e => this.handleChangeDropdownMonth(e));
+    this.dropdownYear.removeEventListener('AXA_CHANGE', e => this.handleChangeDropdownYear(e));
+    window.removeEventListener('click', e => this.handleBodyClick(e));
+  }
+
+  handleViewportCheck(baseElem, elem) {
+    console.log('handle viewport check');
+    if (this.shouldMove(baseElem, elem)) {
+      if (!this.inverted) {
+        this.inverted = true;
+        console.log('handle inverted is set');
+      }
+    } else {
+      this.inverted = false;
+      console.log('handle inverted is unset');
+    }
+  }
+
+  shouldMove(elem) {
+    const elemement = elem.getBoundingClientRect();
+    const moreSpaceOnTopThanBottom = elemement.top > window.innerHeight - elemement.bottom;
+    return moreSpaceOnTopThanBottom;
+  }
+
+  toggleDatepicker() {
+    if (!this.open) {
+      if (window.axaComponents.openDatepickerInstance && window.axaComponents.openDatepickerInstance !== this) {
+        window.axaComponents.openDatepickerInstance.open = false;
+      }
+      this.open = true;
+      window.axaComponents.openDatepickerInstance = this;
+    } else {
+      this.open = false;
+      window.axaComponents.openDatepickerInstance = null;
+    }
+  }
+
+  isValidDate(date) {
+    let out = false;
+    try {
+      const parsedDate = new Date(Date.parse(date));
+      // eslint-disable-next-line no-restricted-properties
+      const isValid = parsedDate instanceof Date && !window.isNaN(parsedDate);
+      const isValidDateLocalized = parseLocalisedDateIfValid(this.locale, date);
+      if ((isValid && isValidDateLocalized) || (!isValid && isValidDateLocalized)) {
+        out = isValidDateLocalized;
+      }
+    } catch (e) {
+      out = false;
+    }
+    return out;
+  }
+
+  render() {
+    return html`
+      <article class="m-datepicker">
+        ${this.inputField &&
+          html`
+            <div class="m-datepicker__input-wrap">
+              <input
+                @keyup="${this.handleInputChange}"
+                class="m-datepicker__input js-datepicker__input"
+                type="text"
+                placeholder="Please select a date"
+                value="${this.outputDate}"
+              />
+              <button class="m-datepicker__input-button" @click="${this.handleInputButtonClick}">
```

Comment:
Carefull, here u are defining a `type='submit` as button. In the absence of a form in not a good idea. Please use `<button type="button">` instead

Reply:
nice catch. done